admin_manual/configuration/server/security/ransomeware-protection.rst

```diff
@@ -0,0 +1,95 @@
+
+Doing so allows all user data that is synchronized with the server to be rolled back to its state before the attack occurred.
+We expect that a combination of Ransomware prevention and protection should reduce risks to a minimum acceptable level.
+
+Other Elements of Ransomware Protection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+==================== ============================================== ================================================
+Name                 Command                                        Description
+==================== ============================================== ================================================
+Ransomguard Scanner  ``occ ransomguard:scan <timestamp> <user>``    A command to scan the ownCloud database for
+                                                                    anomalies in a user account. It enables an administrator to determine the point in time
+                                                                    where undesired actions happened as a
+                                                                    prerequisite for restoration.
+Ransomguard Restorer ``occ ransomguard:restore <timestamp> <user>`` A command for administrators to revert all
+                                                                    operations in a user account that occurred after
+                                                                    a certain point in time.
+==================== ============================================== ================================================
+
+.. note:: 
+   ``<timestamp>`` must be in `the Linux timestamp format`.
+
+Requirements
+~~~~~~~~~~~~
+
+#. **Ransomware Protection.** It records the move operations so that it can revert them in the case of an attack. Other operations are revertible using ownCloud's existing database tables.
+#. **ownCloud Versions App.** It is required to restore older file versions and because capabilities of Ransomware Protection depend on its configuration.
+#. **ownCloud Trash Bin App.** It is required both to restore deleted files and because capabilities of Ransomware Protection depend on its configuration.
+
+Limitations
+~~~~~~~~~~~
+
+-  Ransomware Protection works with master-key based storage encryption. Credential-based storage encryption only Ransomware Prevention (Blocking) works.
+-  Rollback is not based on snapshots.
+-  The `trash bin retention policy`_ may delete files, making them unrecoverable. To avoid this, set ``trashbin\_retention\_obligation`` to ``disabled``, or choose a conservative policy for trash bin retention. However, please be aware that this may increase storage requirements.
+-  Trash bin items may be deleted by the user making them unrecoverable by Ransomware Protection => Users need to know this.
```

Comment:
Somewhere there is an issue asking for a feature setting so that the sysadmin can prevent users from deleting stuff from their trashbin - that might help to ensure old versions of files are available.

Reply:
> Somewhere there is an issue asking for a feature setting so that the sysadmin can prevent users from deleting stuff from their trashbin - that might help to ensure old versions of files are available.

Yep, that would help but it's not yet implemented afaik.

Reply:
Pity.